File: UNIT-1/GCD/euclid.py
# Euclid's algorithm to compute gcd of two numbers

def euclid(m, n):
    while(n != 0):
        r = m % n
        m = n
        n = r
    return m

if(__name__ == "__main__"):
    (a, b) = tuple(map(lambda x: int(x), input("Enter the two numbers:").split()))
    print("GCD=", gcd(a, b))
    

File: UNIT-1/GCD/euclid.c
#include <stdio.h>

int euclid(int m, int n){
    int r;
    while(n != 0){
        r = m % n;
        m = n;
        n = r;
    }
    return m;
}

int main(){
    int a, b;
    printf("Enter the two numbers:");
    scanf("%d %d", &a, &b);
    printf("GCD = %d\n", euclid(a, b));
    return 0;
}